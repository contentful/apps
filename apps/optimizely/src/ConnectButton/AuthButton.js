import React from 'react';
import OptimizelyLogo from '../optimizely-logo';
import { colors } from '../constants';
const AuthButton = (props) => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 200 40" width={200} height={40} {...props}>
      <title>ConnectWithOptimizelyButton</title>
      <rect width="200" height="40" x="0" y="0" rx="6" ry="6" fill={colors.optimizelyBlue} />
      <g transform={'translate(5, 9)'}>
        <OptimizelyLogo width={'25'} height={'25'} arccolor={colors.white} />
      </g>
      <g transform={'translate(32, 22)'}>
        <text x={0} y={5} fill={'white'}>
          {' '}
          Connect with Optimizely{' '}
        </text>
      </g>
    </svg>
  );
};

export default AuthButton;
