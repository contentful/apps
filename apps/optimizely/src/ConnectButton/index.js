import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'emotion';
import { Button } from '@contentful/forma-36-react-components';

// import OptimizelyLogo from './OptimizelyLogo';

// import OptimizelyLogo from '../optimizely-logo';
import AuthButton from './AuthButton';

const styles = {
  connect: css({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    '&:hover': {
      backgroundColor: 'transparent!important', // necessary to eliminate the forma styling in favor of the custom optimizely styling
    },
  }),
};

export default function ConnectButton({ openAuth }) {
  return (
    <Button
      className={styles.connect}
      onClick={openAuth}
      testId="connect-button"
      isFullWidth
      buttonType="naked">
      <AuthButton />
    </Button>
  );
}

ConnectButton.propTypes = {
  openAuth: PropTypes.func.isRequired,
};
