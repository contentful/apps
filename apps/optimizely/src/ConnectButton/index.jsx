import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@emotion/css';
import { Button } from '@contentful/f36-components';

import AuthButton from './AuthButton';

const styles = {
  connect: css({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    '&:hover': {
      backgroundColor: 'transparent!important', // necessary to eliminate the forma styling in favor of the custom optimizely styling
    },
  }),
};

export default function ConnectButton({ openAuth }) {
  return (
    <Button
      className={styles.connect}
      onClick={openAuth}
      testId="connect-button"
      isFullWidth
      variant="transparent">
      <AuthButton />
    </Button>
  );
}

ConnectButton.propTypes = {
  openAuth: PropTypes.func.isRequired,
};
