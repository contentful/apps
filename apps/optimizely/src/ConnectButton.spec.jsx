import React from 'react';
import { cleanup, render, fireEvent, configure } from '@testing-library/react';

import ConnectButton from '../src/ConnectButton';
import { vi } from 'vitest';

configure({ testIdAttribute: 'data-test-id' });

describe('ConnectButton', () => {
  afterEach(cleanup);

  it('should handle the openAuth functionality onClick', () => {
    const openMock = vi.fn();
    const { getByTestId } = render(<ConnectButton openAuth={openMock} />);

    fireEvent.click(getByTestId('connect-button'));

    expect(openMock).toHaveBeenCalledTimes(1);
  });
});
