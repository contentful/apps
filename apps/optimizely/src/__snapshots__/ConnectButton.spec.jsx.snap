// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConnectButton > should match snapshot 1`] = `
<div>
  <button
    class="css-1ev9jpi css-4k45fw"
    data-test-id="connect-button"
    type="button"
  >
    <span
      class="css-40g50j"
    >
      <svg
        height="40"
        viewBox="0 0 200 40"
        width="200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          ConnectWithOptimizelyButton
        </title>
        <rect
          fill="#0037FF"
          height="40"
          rx="6"
          ry="6"
          width="200"
          x="0"
          y="0"
        />
        <g
          transform="translate(5, 9)"
        >
          <svg
            arccolor="#FFFFFF"
            height="25"
            viewBox="0 0 150 150"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M50.29 81.41V99.1a49.41 49.41 0 0 0 49.35-49.35H82a31.71 31.71 0 0 1-31.71 31.66Z"
              style="fill: #3be081;"
            />
            <path
              d="M50.29 130.16a31.43 31.43 0 0 1 0-62.85V49.76a49 49 0 0 0-.14 98h.14Z"
              style="fill: #FFFFFF;"
            />
            <path
              d="M50.29 130.16v17.55a49 49 0 0 0 49-49H81.72a31.46 31.46 0 0 1-31.43 31.45Z"
              style="fill: #0cf;"
            />
            <path
              d="M50.29 32.19v17.56a49 49 0 0 0 49-49H81.72a31.46 31.46 0 0 1-31.43 31.44Z"
              style="fill: #861dff;"
            />
            <path
              d="M99.59 32.19v17.56a49 49 0 0 0 49-49H131a31.46 31.46 0 0 1-31.41 31.44Z"
              style="fill: #fc7f10;"
            />
          </svg>
        </g>
        <g
          transform="translate(32, 22)"
        >
          <text
            fill="white"
            x="0"
            y="5"
          >
             
            Connect with Optimizely
             
          </text>
        </g>
      </svg>
    </span>
  </button>
</div>
`;
